sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -120,6 +120,9 @@ class RocksDB(
       if (conf.resetStatsOnLoad) {
         nativeStats.reset
       }
+      // reset resources to prevent side-effects from previous loaded version
+      prefixScanReuseIter.entrySet().asScala.foreach(_.getValue.close())
+      prefixScanReuseIter.clear()
```

Comment:
nit: How about wrap this 2 lines in a function `closePrefixScan(Reuse?)Iter()`? Since they are already been called in 3 places.

Reply:
Thanks! Addressed via a01d733